### ISSUE TITLE
New nightly version broke `cargo-doc`, unintended use of unstable feature

### ISSUE THREAD

**Issue by marcospb19** [rust-num/num-traits#287]
Crate version: `0.2.16` (latest).
Cargo version: nightly from today (`cargo 1.74.0-nightly (d5336f813 2023-09-14)`).

I'm getting an error when trying to run `cargo doc` in [`a crate`](https://github.com/holmgr/cargo-sweep).

```sh
cargo doc
# or
cargo doc -p num-traits
```

```c
 Documenting num-traits v0.2.16
error[E0658]: custom classes in code blocks are unstable
  --> /home/marcospb19/.cargo/registry/src/index.crates.io-6f17d22bba15001f/num-traits-0.2.16/src/identities.rs:4:1
   |
4  | / /// Defines an additive identity element for `Self`.
5  | | ///
6  | | /// # Laws
7  | | ///
...  |
10 | | /// 0 + a = a       ∀ a ∈ Self
11 | | /// ```
   | |_______^
   |
   = note: see issue #79483 <https://github.com/rust-lang/rust/issues/79483> for more information
   = help: add `#![feature(custom_code_classes_in_docs)]` to the crate attributes to enable
   = note: found these custom classes: class=text

error[E0658]: custom classes in code blocks are unstable
  --> /home/marcospb19/.cargo/registry/src/index.crates.io-6f17d22bba15001f/num-traits-0.2.16/src/identities.rs:80:1
   |
80 | / /// Defines a multiplicative identity element for `Self`.
81 | | ///
82 | | /// # Laws
83 | | ///
...  |
86 | | /// 1 * a = a       ∀ a ∈ Self
87 | | /// ```
   | |_______^
   |
   = note: see issue #79483 <https://github.com/rust-lang/rust/issues/79483> for more information
   = help: add `#![feature(custom_code_classes_in_docs)]` to the crate attributes to enable
   = note: found these custom classes: class=text

For more information about this error, try `rustc --explain E0658`.
error: could not document `num-traits`
```

Therefore, I cannot see the docs for my crate.

**Comment by kangalio** [comment]
Also hitting this in ihttps://github.com/serenity-rs/poise/pull/204

**Comment by cuviper** [comment]
rust-lang/rust#115914 relaxed this to a warning, shipped in `rustc 1.74.0-nightly (65ea825f4 2023-09-18)`. Furthermore, that seems to now respect cap-lints, so you shouldn't even see the warning here unless you use `cargo doc -v -v`.
